### PR TITLE
tips from db

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

File: modules/cache/logic/cacheLogic.js [hunk 1]
```diff
@@ -366,59 +334,8 @@ const CacheLogic = {
...
-        url,
-        ...CacheLogic.statsForTips(groupedByUrl[url]),
-      }));
-
-      return {
-        ...CacheLogic.statsForTips(tips),
-        by_url: statsByUrl,
-      };
-    }, cache.longCacheTime);
+    throw Error('no more stats from cache');
```
Comment (kenodressel) [inline modules/cache/logic/cacheLogic.js:338]: I guess we can remove this function now

File: modules/cache/utils/cache.js [hunk 2]
```diff
@@ -96,8 +97,10 @@ cache.keepHot = keepHotFunction => {
     logger.info(`cache keepHot ${new Date().getTime() - start}ms`);
   });
 
-  keepHotLogic();
-  setInterval(keepHotLogic, cache.keepHotInterval);
+  queueLogic.subscribeToMessage(MESSAGE_QUEUES.SCHEDULED_EVENTS, MESSAGES.SCHEDULED_EVENTS.COMMANDS.CACHE_KEEPHOT, async message => {
```
Comment (kenodressel) [inline modules/cache/utils/cache.js:100]: it feels weird that the util subscribes to the scheduler, should we move this to `CacheLogic`?

File: modules/comment/tests/comments.js [hunk 3]
```diff
@@ -34,56 +34,32 @@ describe('Comments', () => {
       object => Comment.update({ parentId: null }, { where: { id: object.id } }),
     ));
```
Comment (kenodressel) [inline modules/comment/tests/comments.js:36]: use `fakeTipsAndUpdateDB`

File: modules/queue/logic/schedulerLogic.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+const scheduledEvents = [
+  { message: MESSAGES.SCHEDULED_EVENTS.COMMANDS.CACHE_KEEPHOT, interval: process.env.KEEP_HOT_INTERVAL || 5 * msInMin, onceAtStartup: true },
+  { message: MESSAGES.SCHEDULED_EVENTS.COMMANDS.UPDATE_CHAIN_NAMES, interval: 10 * msInMin, onceAtStartup: true },
+  { message: MESSAGES.SCHEDULED_EVENTS.COMMANDS.UPDATE_TIPS_RETIPS_CLAIMS, interval: 10 * msInMin, onceAtStartup: true },
+  { message: MESSAGES.SCHEDULED_EVENTS.COMMANDS.UPDATE_EVENTS, interval: 10 * msInMin, onceAtStartup: true },
+];
+
+const SchedulerLogic = {
+  init() {
+    setTimeout(() => scheduledEvents.forEach(async ({ message, interval, onceAtStartup }) => {
```
Comment (kenodressel) [inline modules/queue/logic/schedulerLogic.js:15]: why wait for a second before this?

File: utils/testingUtil.js [hunk 5]
```diff
@@ -75,6 +80,45 @@ const performSignedGETRequest = (server, url, privateKey = null) => new Promise(
     });
 });
 
+const fakeTipsAndUpdateDB = dbsToClear => async (fakeData, clearData = true) => {
```
Comment (kenodressel) [inline utils/testingUtil.js:83]: rename this to something more suiting

**Review by kenodressel — CHANGES_REQUESTED** [review]

File: migrations/20210218-additional-tip-fields.js [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+                {
+                    transaction: transaction
+                }
+            ]
+        },
+        {
+            fn: "addColumn",
+            params: [
+                "Tips",
+                "receiver",
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply migrations/20210218-additional-tip-fields.js:86]: yes

File: migrations/20210218-add-tip-title-and-topics.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+/**
+ * Actions summary:
+ *
+ * addColumn "topics" to table "Tips"
+ * addColumn "title" to table "Tips"
+ *
+ **/
+
+var info = {
+    "revision": 11,
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply migrations/20210218-add-tip-title-and-topics.js:14]: yes

[the remark this answers is not in the document]
Comment (thepiwo) [reply migrations/20210218-add-tip-title-and-topics.js:14]: doesn't matter, I think

File: migrations/20210218-additional-tip-fields.js [hunk 8]
```diff
@@ -0,0 +1,286 @@
...
+        if (this.useTransaction) {
+            return queryInterface.sequelize.transaction(run);
+        } else {
+            return run(null);
+        }
+    },
+    up: async function(queryInterface, Sequelize)
+    {
+      const transaction = await queryInterface.sequelize.transaction();
+      await queryInterface.sequelize.query('ALTER TYPE "enum_Tips_type" ADD VALUE \'POST_WITHOUT_TIP\';');
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply migrations/20210218-additional-tip-fields.js:271]: have to check for this in general again

File: modules/aeternity/utils/tipTopicUtil.js [hunk 9]
```diff
@@ -1,9 +1,8 @@
 const BigNumber = require('bignumber.js');
-const Util = require('./util');
 
 const topicsRegex = /(#[a-zA-Z]+\b)(?!;)/g;
 
-const getTipTopics = tips => {
+const getTipTopics = tips => { // TODO move to db
```
[the remark this answers is not in the document]
Comment (thepiwo) [reply modules/aeternity/utils/tipTopicUtil.js:5]: have to check

File: modules/cache/logic/cacheLogic.js (hunk 1 above)
Comment (thepiwo) [reply modules/cache/logic/cacheLogic.js:338]: yes

File: modules/cache/utils/cache.js (hunk 2 above)
Comment (thepiwo) [reply modules/cache/utils/cache.js:100]: fine for me, I can adjust it

File: modules/comment/tests/comments.js (hunk 3 above)
Comment (thepiwo) [reply modules/comment/tests/comments.js:36]: I don't understand

File: modules/queue/logic/schedulerLogic.js (hunk 4 above)
Comment (thepiwo) [reply modules/queue/logic/schedulerLogic.js:15]: so all listeners for these events have finished registering for sure

File: utils/testingUtil.js (hunk 5 above)
Comment (thepiwo) [reply utils/testingUtil.js:83]: can you provide a PR

File: modules/queue/logic/schedulerLogic.js (hunk 4 above)
Comment (kenodressel) [reply modules/queue/logic/schedulerLogic.js:15]: did you experience any issues with this? or is it a precautionary measure?

File: utils/testingUtil.js (hunk 5 above)
Comment (kenodressel) [reply utils/testingUtil.js:83]: https://github.com/aeternity/tipping-community-backend/pull/312

File: migrations/20210218-add-tip-title-and-topics.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (kenodressel) [reply migrations/20210218-add-tip-title-and-topics.js:14]: https://github.com/aeternity/tipping-community-backend/pull/313

File: modules/cache/logic/cacheLogic.js (hunk 1 above)
Comment (kenodressel) [reply modules/cache/logic/cacheLogic.js:338]: https://github.com/aeternity/tipping-community-backend/pull/314

File: modules/queue/logic/schedulerLogic.js (hunk 4 above)
Comment (thepiwo) [reply modules/queue/logic/schedulerLogic.js:15]: I think initially I had some issues, but I didn't check it again, I'll do

File: migrations/20210218-additional-tip-fields.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (thepiwo) [reply migrations/20210218-additional-tip-fields.js]: should be all fine now

File: modules/queue/logic/schedulerLogic.js (hunk 4 above)
Comment (thepiwo) [reply modules/queue/logic/schedulerLogic.js]: It's not needed, already removed it

File: modules/cache/utils/cache.js (hunk 2 above)
Comment (thepiwo) [reply modules/cache/utils/cache.js]: all keephot is now moved to cache logic

File: modules/aeternity/utils/tipTopicUtil.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (thepiwo) [reply modules/aeternity/utils/tipTopicUtil.js:5]: lets have a call about this

File: modules/cache/logic/cacheLogic.js [hunk 10]
```diff
@@ -17,8 +17,7 @@ const logger = require('../../../utils/logger')(module);
 
 const CacheLogic = {
   init() {
-    // INIT ONCE
-    cache.setKeepHot(CacheLogic.keepHotFunction);
+    CacheLogic.keepHot(CacheLogic.keepHotFunction);
   },
 
   async keepHotFunction() {
```
Comment (kenodressel) [inline modules/cache/logic/cacheLogic.js:23]: why pass it as an argument? lets hardcode it or at least move it out of the init function

**Review by kenodressel — APPROVED** [review]

**Pull request merged**